hypothesis-python/src/hypothesis/searchstrategy/misc.py

```diff
@@ -154,3 +120,26 @@ def check_index(i):
         # If there are no allowed indices, the filter couldn't be satisfied.
 
         return filter_not_satisfied
+
+
+class JustStrategy(SampledFromStrategy):
+    """A strategy which always returns a single fixed value.
+
+    It's implemented as a length-one SampledFromStrategy so that all our
+    special-case logic for filtering and sets applies also to just(x).
+    """
+
+    def __init__(self, value):
+        SampledFromStrategy.__init__(self, [value])
+
+    def __repr__(self):
+        return "just(%r)" % (self.elements[0],)
```

Comment:
Would it be worth adding the following code?

```python
@property 
def value(self):
    return self.elements[0]
```

You end up using `elements[0]` in a bunch of places so it's a natural convenience, and by adding this property you don't need to change as much of the other code.

Reply:
My own sense of style is to leave it as `elements[0]`; it's clear enough inside the class and there's only a single use outside it.  So it feels cleaner to me to avoid duplicating the information :man_shrugging:

Reply:
I don't have a strong opinion in favour of the property, but I'm a little confused by the idea that extracting a common method is *duplicating* information!

Reply:
:man_facepalming:, skipped right over the code block and was thinking of dropping `self.value = value` back into `__init__`.

Reply:
I had assumed from the last comment that you were planning to implement this change. Is that not the case? Fine if not, but I was waiting on that to do a second review.

Reply:
Whoops, that comment really isn't clear... if you're happy to merge it as-is, so am I.

Reply:
Let's say "willing". I have a mild preference for the change, but if you're opposed I'm happy to go with your inclination if you're against it.

Reply:
Mild preference is to decide me; I'll push a commit with the property once the current build is done.